### ISSUE TITLE
Push Step not Authorized to Push to OpenShift Image Registry

### ISSUE THREAD

**Issue by danielhelfand** [openshift-labs/lab-openshift-pipelines-with-tekton#15]
The push step of the `s2i-java-8` task used in this workshop fails with the following error:

```
[build : build-step-push] unauthorized: authentication required
```

This is possibly due to the service account used in the tutorial not having appropriate permissions. This step has worked in the past outside of homeroom and could be specific to how homeroom creates users for that allow end users to go through the workshop.

**Comment by danielhelfand** [comment]
To solve this, two things must be done:

1. The service account `pipeline` needs the following permissions:

```
$ oc adm policy add-scc-to-user privileged -z pipeline
$ oc adm policy add-role-to-user edit -z pipeline
```

2. The image registry resource for the tutorial needs to be specific to the namespace. Right now, it is set up as `pipelines tutorial`, but it must correspond the namespace generated by Homeroom:

https://github.com/openshift-labs/lab-openshift-pipelines-with-tekton/blob/master/code/exercise/petclinic-image-pipeline-resource.yaml#L11

**Comment by danielhelfand** [comment]
Issue was with the service account needing to push to the correct namespace in the registry. Addressed by #24.
